### PR TITLE
DOCS-#3792, DOCS-#2410: Improve "Getting Started" section

### LINKED ISSUE #3792: Improve structure for "Getting Started" section
Related to #3687. Suggesting new structure for "Getting Started" section, which is intended as a starting point for Modin users.
 
```
Getting Started/
├─ *Quickstart Guide (minimal install & setup script, 1-line replacement, mini-benchmark to demo speedup, same as updated README)
├─ Installation (pip, conda, source)/
├─ *Example (link to 1-2 notebook ex of how Modin is used for XYZ)
├─ *Tutorial (link to tutorial notebooks)
├─ Using Modin/
│  ├─ Using Modin locally
│  ├─ Using Modin in a cluster
├─ FAQs
├─ Troubleshooting
├─ Why use Modin?/
│  ├─ Modin vs. pandas
│  ├─ Out of Core in Modin
│  ├─ Modin vs. other distributed dataframes (disclaimer/overview)
│  ├─│  ├─ Modin vs. Dask DataFrame (#3779)
│  ├─│  ├─ Modin vs. Koalas
```
`*` indicates that the documentation page would have an associated notebook.
(edited based on thread below)

Currently, the tutorial notebooks are [here](https://github.com/modin-project/modin/blob/master/examples/tutorial/tutorial_notebooks), which is not linked in the current docs (see #2412).

Note that this PR will not include new files for:
```
│  ├─ FAQs
│  ├─ Troubleshooting
```
given that it will be part of #2635.

Beyond the Getting Started Section, the overall top-level guide should be: 

```
Getting Started
Usage Guide
Supported APIs
Developer
Contact
```

### PR DESCRIPTION
**Pull request by dorisjlee**
Major refactoring for "Getting Started section". Migrated from PR#3898.

- Improved flow and organization of sections based on #3792 structure
- Updated Using Modin locally and in cluster setting
- New quickstart guide and corresponding notebook
- New examples page linking to notebook tutorials, articles, and talks about Modin
- Added estimated read time on each page

Preview: https://modin--3965.org.readthedocs.build/en/3898/getting_started/quickstart.html

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3792, #2410
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/getting_started/examples.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
+Examples and Resources 
+========================
+
+The earlier sections have covered the basics of using Modin to speed up your pandas workflow. Here you can find additional resources to learn about Modin. To learn more about advanced usage for Modin, please refer to :doc:`this section </advanced_usage/index>`.
```
Comment (devin-petersohn) [inline docs/getting_started/examples.rst:4]: Which earlier sections?

File: docs/getting_started/faq.rst [hunk 2]
```diff
@@ -111,6 +117,42 @@ We also have an experimental OmniSciDB-based engine of Modin you can read about
 We plan to support more execution engines in future. If you have a specific request, 
 please post on the #feature-requests channel on our Slack_ community. 
 
+How do I connect Modin to a database via `read_sql`?
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```
Comment (devin-petersohn) [inline docs/getting_started/faq.rst:121]:
```suggestion
How do I connect Modin to a database via `read_sql`?
""""""""""""""""""""""""""""""""""""""""""""""""""""
```

File: docs/getting_started/faq.rst [hunk 3]
```diff
@@ -88,7 +94,7 @@ import with Modin import:
     import modin.pandas as pd
 
 Which execution engine (Ray or Dask) should I use for Modin?
-""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```
Comment (devin-petersohn) [inline docs/getting_started/faq.rst:97]:
```suggestion
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```

File: docs/getting_started/troubleshooting.rst [hunk 4]
```diff
@@ -196,6 +165,55 @@ funcion for example) as it is shown in the example below:
...
+**Solution**
+
+.. code-block:: python
+
+  import modin.pandas as pd
+  pd.DEFAULT_NPARTITIONS = 2 * pd.DEFAULT_NPARTITIONS
+
+This will set the number of partitions to a higher count, and reduce the size in each.
+If this does not work for you, please open an issue_.
+
```
Comment (devin-petersohn) [inline docs/getting_started/troubleshooting.rst:216]:
```suggestion
```

These two issues don't happen anymore, they have been fixed.

File: docs/getting_started/using_modin/using_modin_locally.rst [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+  # import pandas as pd
+  import modin.pandas as pd
+
+**That's it. You're ready to use Modin on your previous pandas workflows!** 
+
+Optional Configurations
+-------------------------
+
+When using Modin locally on a single machine or laptop (without a cluster), Modin will automatically create and manage a local Dask or Ray cluster for the executing your code. So when you run an operation for the first time with Modin, you will see a message like this, indicating that a Modin has automatically initialized a cluster for you:
```
Comment (devin-petersohn) [inline docs/getting_started/using_modin/using_modin_locally.rst:29]:
```suggestion
When using Modin locally on a single machine or laptop (without a cluster), Modin will automatically create and manage a local Dask or Ray cluster for the executing your code. So when you run an operation for the first time with Modin, you will see a message like this, indicating that a Modin has automatically initialized a local cluster for you:
```

File: docs/getting_started/why_modin/pandas.rst [hunk 6]
```diff
@@ -1,13 +1,15 @@
-Modin vs. pandas
-================
+How Modin differs from pandas?
+===============================
```
Comment (devin-petersohn) [inline docs/getting_started/why_modin/pandas.rst:2]:
```suggestion
How does Modin differ from pandas?
==================================
```

**Review by devin-petersohn — COMMENTED** [review]
Thanks @dorisjlee for this!

One minor thing, if it's not too much trouble: break longer lines at around 88 characters so that in the future the diffs are smaller and merges are easier if/when we need to make changes.

File: docs/getting_started/quickstart.rst [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+    examples
+    faq
+    troubleshooting
+
+.. meta::
+    :description lang=en:
+        Introduction to Modin.
+
+Quick Start Guide
+------------------
```
Comment (YarShev) [inline docs/getting_started/quickstart.rst:25]: Please remove extra symbols where necessary.

File: docs/getting_started/examples.rst [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+---------------
+
+The following notebooks demonstrate how Modin can be used for scalable data science: 
+
+- Quickstart Guide to Modin [`Source <https://github.com/modin-project/modin/tree/master/examples/quickstart.ipynb>`_]
+- Using Modin with the NYC Taxi Dataset [`Source <https://github.com/modin-project/modin/blob/master/examples/jupyter/NYC_Taxi.ipynb>`_] 
+- Using Experimental Modin in the cloud with the NYC Taxi Dataset on a AWS cluster [`Source <https://github.com/modin-project/modin/blob/master/examples/jupyter/NYC_Taxi_cloud.ipynb>`_] 
+- Modin for Machine Learning with scikit-learn [`Source <https://github.com/modin-project/modin/blob/master/examples/modin-scikit-learn-example.ipynb>`_] 
+
+Tutorials
```
Comment (YarShev) [inline docs/getting_started/examples.rst:16]: I was not able to get to no one of the exercises. Please check the links.

**Comment by Garra1980** [comment]
can we insert somewhere (probably in Using Modin Locally) a Note about impossibility of mixing different distribution mechanisms (eg Ray and python multiprocessing)?

File: docs/getting_started/quickstart.rst (hunk 7 above)
Comment (dorisjlee) [reply docs/getting_started/quickstart.rst:25]: Fixed.

File: docs/getting_started/quickstart.rst [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+  | You can follow along this tutorial in a Jupyter notebook `here <https://github.com/modin-project/modin/tree/master/examples/quickstart.ipynb>`_. 
+
+.. toctree::
+    :hidden:
+    :maxdepth: 4
+    
+    10-min Quickstart Guide <self>
+    installation
+    using_modin/using_modin
+    why_modin/why_modin
```
[the remark this answers is not in the document]
Comment (dorisjlee) [reply docs/getting_started/quickstart.rst:15]: `using_modin` is more important for the user in the "Getting Started" section than `why_modin` which is more around the details of how Modin works and why it's different than existing solutions.

File: docs/getting_started/examples.rst (hunk 1 above)
Comment (dorisjlee) [reply docs/getting_started/examples.rst:4]: Removed.

File: docs/getting_started/examples.rst (hunk 8 above)
Comment (dorisjlee) [reply docs/getting_started/examples.rst:16]: Nice catch! Fixed now!

**Comment by dorisjlee** [comment]
> One minor thing, if it's not too much trouble: break longer lines at around 88 characters so that in the future the diffs are smaller and merges are easier if/when we need to make changes.

Added line breaks at 88 chars

**Comment by dorisjlee** [comment]
> can we insert somewhere (probably in Using Modin Locally) a Note about impossibility of mixing different distribution mechanisms (eg Ray and python multiprocessing)?

Added note [here](https://github.com/modin-project/modin/pull/3965/files#diff-49fcfa4583cb683eb2adf0a65779e5fbfcff4cbafa210b50a7e095d087e5f8f7R73)

File: docs/getting_started/using_modin/using_modin_locally.rst [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+
+      client = Client()
+
+Finally, if you already have an Ray or Dask engine initialized, Modin will 
+automatically attach to whichever engine is available. If you are interested in using 
+Modin with OmniSci engine, please refer to :doc:`these instructions </developer/using_omnisci>`. For additional information on other settings you can configure, see 
+:doc:`this page </flow/modin/config>` for more details.
+
+.. note:: 
+   Note that Modin can not be used in conjunction with other distribution mechanism.
```
Comment (YarShev) [inline docs/getting_started/using_modin/using_modin_locally.rst:74]: This is a highly general phrase. Ray can be used with Dask, for example, if Ray and Dask utilize enough resources for their execution. I would rephrase this phrase a bit and more focused on python multiprocessing. Also, I think this should be put in troubleshooting page.

**Review by YarShev — COMMENTED** [review]
Left a comment. The rest changes are fine to me.

File: docs/getting_started/using_modin/using_modin_locally.rst (hunk 10 above)
Comment (dorisjlee) [reply docs/getting_started/using_modin/using_modin_locally.rst:74]: I moved this to the troubleshooting section and specified that this is only for python multiprocessing. I can not actually reproduce any error or warning when using Modin on Ray and python multiprocessing together, so I'm not sure what this is referring to.

File: docs/getting_started/troubleshooting.rst [hunk 11]
```diff
@@ -196,6 +165,15 @@ funcion for example) as it is shown in the example below:
   pd_df = pd_df.set_index(index_col_name)
   pd_df.index.name = None
 
+
+Using Modin with python multiprocessing
+"""""""""""""""""""""""""""""""""""""""
+
+Note that Modin can not be used in conjunction with python multiprocessing. For example, if the Ray engine is selected, it can not be used alongside python multiprocessing.
```
Comment (YarShev) [inline docs/getting_started/troubleshooting.rst:172]:
```suggestion
We strongly recommend not to mix the use of Modin with Ray or Dask engine selected
in conjunction with python multiprocessing because that can lead to undefined behavior.
One of such examples is shown below:

.. code-block:: python
  import modin.pandas as pd
  
  # Ray engine is used by default
  df = pandas.DataFrame([1, 2, 3])
  
  def f(arg):
    return df + arg

  if __name__ == '__main__':
    from multiprocessing import Pool
    
    with Pool(5) as p:
        print(p.map(f, [1]))

Even if this example may work on your machine, we do not recommend similar scenarios.
The python multiprocessing will cause conflicts with excessive resource use
by launching duplicated Ray clusters on the same machine.
```

**Comment by Garra1980** [comment]
Btw, do we have Release Notes at some place or I missed them? If we don't have such topic - how do all think about adding it?

**Comment by devin-petersohn** [comment]
@Garra1980 it is a great suggestion. Let's add it in a new issue

File: docs/getting_started/troubleshooting.rst (hunk 11 above)
Comment (dorisjlee) [reply docs/getting_started/troubleshooting.rst]: This looks good, thanks @YarShev !

**Comment by Garra1980** [comment]
> Release notes are [here](https://github.com/modin-project/modin/releases). Instructions on how to write the release notes should probably go in the Contributing section and not this section.

Yeah, I was talking about adding link on Release Notes not on instructions how to write them :)

File: docs/getting_started/examples.rst [hunk 12]
```diff
@@ -0,0 +1,67 @@
+Examples and Resources 
+======================
+
+Here you can find additional resources to learn about Modin. To learn more about 
+advanced usage for Modin, please refer to :doc:`this section </advanced_usage/index>`.
+
+Usage Examples
+---------------
```
Comment (YarShev) [inline docs/getting_started/examples.rst:8]:
```suggestion
Usage Examples
--------------
```

File: docs/getting_started/examples.rst [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+
+The following notebooks demonstrate how Modin can be used for scalable data science: 
+
+- Quickstart Guide to Modin [`Source <https://github.com/modin-project/modin/tree/master/examples/quickstart.ipynb>`_]
+- Using Modin with the NYC Taxi Dataset [`Source <https://github.com/modin-project/modin/blob/master/examples/jupyter/NYC_Taxi.ipynb>`_] 
+- Using Experimental Modin in the cloud with the NYC Taxi Dataset on a AWS cluster [`Source <https://github.com/modin-project/modin/blob/master/examples/jupyter/NYC_Taxi_cloud.ipynb>`_] 
+- Modin for Machine Learning with scikit-learn [`Source <https://github.com/modin-project/modin/blob/master/examples/modin-scikit-learn-example.ipynb>`_] 
+
+Tutorials  
+----------
```
Comment (YarShev) [inline docs/getting_started/examples.rst:18]:
```suggestion
Tutorials  
---------
```

File: docs/getting_started/examples.rst [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+
+
+The following tutorials covers more advanced features in Modin: 
+
+- Exercise 4: Experimental Features in Modin (Spreadsheet, Progress Bar) [`Source <https://github.com/modin-project/modin/blob/721d2d2fe2a46d3b7442e1ab2c7b70ad3bff0712/examples/tutorial/tutorial_notebooks/introduction/exercise_4.ipynb>`_]
+- Exercise 5: Setting up Modin in a Cluster Environment [`Source <https://github.com/modin-project/modin/blob/721d2d2fe2a46d3b7442e1ab2c7b70ad3bff0712/examples/tutorial/tutorial_notebooks/cluster/exercise_5.ipynb>`_]
+- Exercise 6: Running Modin in a Cluster Environment [`Source <https://github.com/modin-project/modin/blob/721d2d2fe2a46d3b7442e1ab2c7b70ad3bff0712/examples/tutorial/tutorial_notebooks/cluster/exercise_6.ipynb>`_]
+
+Talks & Podcasts
+-----------------
```
Comment (YarShev) [inline docs/getting_started/examples.rst:34]:
```suggestion
Talks & Podcasts
----------------
```

File: docs/getting_started/faq.rst [hunk 15]
```diff
@@ -5,6 +5,9 @@ Below, you will find answers to the most commonly asked questions about
 Modin. If you still cannot find the answer you are looking for, please post your
 question on the #support channel on our Slack_ community or open a Github issue_.
 
+FAQs: Why choose Modin?
+------------------------
```
Comment (YarShev) [inline docs/getting_started/faq.rst:9]:
```suggestion
FAQs: Why choose Modin?
-----------------------
```

File: docs/getting_started/faq.rst [hunk 16]
```diff
@@ -67,6 +70,9 @@ Ray or Dask to execute computation, and then return the results to the user.
 
 For more details, take a look at our system :doc:`architecture </developer/architecture>`. 
 
+FAQs: How to use Modin?
+------------------------
```
Comment (YarShev) [inline docs/getting_started/faq.rst:74]:
```suggestion
FAQs: How to use Modin?
-----------------------
```

File: docs/getting_started/troubleshooting.rst [hunk 17]
```diff
@@ -11,50 +11,19 @@ Frequently encountered issues
...
-working on remote tasks. This is working as intended, as it is not yet possible in Ray
-to kill a task that has already started computation.
-
-The other common way this ``Error`` is encountered is to let your computer go to sleep.
-As an optimization, Ray will shutdown whenever the computer goes to sleep. This will
-result in the same issue as above, because there is no longer a running instance of the
-plasma store.
+Warning during execution: ``defaulting to pandas``
+"""""""""""""""""""""""""""""""""""""""""""""""""""
+Please note, that while Modin covers a large portion of the pandas API, not all functionality is implemented. For methods that are not yet implemented, such as ``asfreq``, you may see the following:
```
Comment (YarShev) [inline docs/getting_started/troubleshooting.rst:16]:
```suggestion
Warning during execution: ``defaulting to pandas``
""""""""""""""""""""""""""""""""""""""""""""""""""

Please note, that while Modin covers a large portion of the pandas API, not all functionality is implemented. For methods that are not yet implemented, such as ``asfreq``, you may see the following:
```

File: docs/getting_started/troubleshooting.rst [hunk 18]
```diff
@@ -196,6 +165,36 @@ funcion for example) as it is shown in the example below:
...
+    
+    with Pool(5) as p:
+        print(p.map(f, [1]))
+
+Even if this example may work on your machine, we do not recommend similar scenarios.
+The python multiprocessing will cause conflicts with excessive resource use
+by launching duplicated Ray clusters on the same machine.
+
+Common errors
+---------------
```
Comment (YarShev) [inline docs/getting_started/troubleshooting.rst:196]:
```suggestion
Common errors
-------------
```

File: docs/getting_started/using_modin/using_modin.rst [hunk 19]
```diff
@@ -0,0 +1,12 @@
+Using Modin
+===============
```
Comment (YarShev) [inline docs/getting_started/using_modin/using_modin.rst:2]:
```suggestion
Using Modin
===========
```

File: docs/getting_started/using_modin/using_modin_cluster.rst [hunk 20]
```diff
@@ -0,0 +1,113 @@
+==========================
+Using Modin in a Cluster
+==========================
```
Comment (YarShev) [inline docs/getting_started/using_modin/using_modin_cluster.rst:3]:
```suggestion
========================
Using Modin in a Cluster
========================
```

File: docs/getting_started/using_modin/using_modin_cluster.rst [hunk 21]
```diff
@@ -0,0 +1,113 @@
...
+
+   ray up modin-cluster.yaml
+
+This configuration script starts 1 head node (m5.24xlarge) and 7 workers (m5.24xlarge), 
+768 total CPUs. For more information on how to launch a Ray cluster across different 
+cloud providers or on-premise, you can also refer to the Ray documentation `here <https://docs.ray.io/en/latest/cluster/cloud.html>`_.
+
+
+Connecting to a Ray Cluster
+-----------------------------
```
Comment (YarShev) [inline docs/getting_started/using_modin/using_modin_cluster.rst:38]:
```suggestion
Connecting to a Ray Cluster
---------------------------
```

File: docs/getting_started/using_modin/using_modin_cluster.rst [hunk 22]
```diff
@@ -0,0 +1,113 @@
...
+chart shows the performance of the above operations with 2, 4, and 8 nodes, with 
+improvements in performance as we increase the number of resources Modin can use.
+
+.. image:: ../../../examples/tutorial/tutorial_notebooks/img/modin_cluster_perf.png
+   :alt: Cluster Performance
+   :align: center
+   :scale: 90%
+
+Advanced: Configuring your Ray Environment
+-------------------------------------------
```
Comment (YarShev) [inline docs/getting_started/using_modin/using_modin_cluster.rst:93]:
```suggestion
Advanced: Configuring your Ray Environment
------------------------------------------
```

File: docs/getting_started/using_modin/using_modin_locally.rst [hunk 23]
```diff
@@ -0,0 +1,119 @@
+=====================
+Using Modin Locally
+=====================
```
Comment (YarShev) [inline docs/getting_started/using_modin/using_modin_locally.rst:3]:
```suggestion
===================
Using Modin Locally
===================
```

File: docs/getting_started/using_modin/using_modin_locally.rst [hunk 24]
```diff
@@ -0,0 +1,119 @@
...
+
+.. code-block:: python
+
+  # import pandas as pd
+  import modin.pandas as pd
+
+**That's it. You're ready to use Modin on your previous pandas workflows!** 
+
+Optional Configurations
+-------------------------
```
Comment (YarShev) [inline docs/getting_started/using_modin/using_modin_locally.rst:26]:
```suggestion
Optional Configurations
-----------------------
```

File: docs/getting_started/using_modin/using_modin_locally.rst [hunk 25]
```diff
@@ -0,0 +1,119 @@
...
+
+      client = Client()
+
+Finally, if you already have an Ray or Dask engine initialized, Modin will 
+automatically attach to whichever engine is available. If you are interested in using 
+Modin with OmniSci engine, please refer to :doc:`these instructions </developer/using_omnisci>`. For additional information on other settings you can configure, see 
+:doc:`this page </flow/modin/config>` for more details.
+
+Advanced: Configuring the resources Modin uses
+-----------------------------------------------
```
Comment (YarShev) [inline docs/getting_started/using_modin/using_modin_locally.rst:74]:
```suggestion
Advanced: Configuring the resources Modin uses
----------------------------------------------
```

File: docs/getting_started/why_modin/why_modin.rst [hunk 26]
```diff
@@ -0,0 +1,19 @@
+Why Modin?
+===============
```
Comment (YarShev) [inline docs/getting_started/why_modin/why_modin.rst:2]:
```suggestion
Why Modin?
==========
```

**Comment by YarShev** [comment]
@dorisjlee, I removed redundant symbols in titles. Hope, you are fine with this.

**Review by YarShev — DISMISSED** [review]
LGTM

**Comment by dorisjlee** [comment]
Thanks @YarShev!

**Comment by Garra1980** [comment]
> @Garra1980 it is a great suggestion. Let's add it in a new issue

@dorisjlee could you please create an issue for Release Notes?

**Comment by dorisjlee** [comment]
Sure @Garra1980! I've now added this as #4024

**Review by devin-petersohn — APPROVED** [review]
Thanks @dorisjlee, LGTM

**Review by YarShev — APPROVED** [review]

**Comment by Garra1980** [comment]
Thanks a lot @dorisjlee!!
